### DISCUSSION TITLE
Option to hide file extension in F2 inline rename

### DISCUSSION THREAD

**Discussion by advice2020** [doublecmd/doublecmd#1134] [Suggestions]
### Description
I was wondering if it would be considered to offer an option to be able to hide file extensions when using the F2 inline rename in file panes.

Personally, I feel like extensions can often get in the way when renaming filenames.  Somewhat uncommon to have to rename a files extension, unless turning txt files into scripts.
Not showing extension all the time would also allow user to be able to hit the "End" key to go to the end of the filename instead of going to the end of the extension.
Also if user uses the "Name Without Extension" column it can be strange that when you enter the inline rename mode (F2) that all of a sudden the extension shows up after filename.

The idea mentioned in the proposed section below would allow the file extension to only be shown if needed by user.

### Proposed technical implementation details
DC contains the following setting in Options - File Operations
`Select file name without extension when renaming`

I am suggesting to add a conditional checkbox sub-setting underneath it that becomes available when that main setting is checked/"on".
This new setting would read something similar to
'Hide extension for first inline rename keypress'

When this new sub-setting is checked/"on" it would perform like the following example

`EXAMPLE:`
Example file
Apple Pie.txt

If you are using the "Name Without Extension" column it shows
Apple Pie

-When you hit inline rename hotkey (F2 default) it would show the filename exactly how it is shown in the column in the filepane without the extension, now highlighted
`Apple Pie`

-Hitting hotkey a 2nd time would show the extension, highlighting both the filename + extension
'Apple Pie.txt'
(I would prefer this 2nd hotkey cycle to highlight only the extension Apple.`txt` but I mentioned the above to keep consistant with how DC F2 cycle performs)

-Hitting hotkey a 3rd time would cycle to showing the extension highlighted
Apple Pie.`txt`

Hopefully this will be considered to be added.
Thank You

`Edit:`
Another thing I did not mention is that when extensions are shown, you cannot bring cursor to a location in filename and then use `Shift+End` key to apply highlight from the cursor location to end of filename to remove or type in new text.  When you apply this hotkey it goes to the end of the extension instead.

**Comment by Skif-off** [comment]
Options > File operations > "Select file name without extension when renaming"?

**Comment by advice2020** [comment]
Thank you for the response
I am aware of this option but if possible would be nice to hide the extension from view and only be seen when F2 key is hit a second time.

**Comment by JonnyTech** [comment]
Is the multiple F2 keypress not suitable?
Press F2 = highlight filename and extension
Press F2 again = highlight filename only
Then enter new filename without extension

**Comment by advice2020** [comment]
@JonnyTech 
Thank you for your response.
Actually I am currently using the following setting....
`Options - File Operations`
`User Interface - Select file name without extension when renaming`
Which does what you are mentioning, it is ok but
I do not want to see the extension at all.

**Comment by Skif-off** [comment]
> The F2 cycle this program used was…

This is not true, Double Commander uses several separators, not just a dot. See [Renaming files](https://doublecmd.github.io/doc/en/help.html#rename_files).

> The weird thing is, in Linux the extensions do not mean anything (MIME wise)

This is not true too, see `freedesktop.org.xml.in` or `freedesktop.org.xml` in the [shared-mime-info](https://www.freedesktop.org/wiki/Software/shared-mime-info/)

**Comment by advice2020** [comment]
I have decided to update my original post if anyone is interested in reading it at the top of the page.

@Skif-off
Thank you for your response, the line above the line that you quoted from my post mentioned that I am referring to a different FM when I said "this program" I was not referring to DC, so it is true :)
But no longer relevant because as mentioned I just updated my main post.

**Comment by advice2020** [comment]
I added an edit to my thread post at top of this page with another benefit of adding something like this, if anyone is interested.

------------------------------------

I apologize to keep reviving this thread but this is one of my requests that I desire the most to be added.
Not sure why I never thought of this before, but if none of the ideas mentioned above are considered, what about adding this to the `cm_RenameOnly` command that is assigned to F2 key by default.

`CURRENT PARAMETER`
-Show filename + extension (which is the current default)

`NEW PARAMETER`
-Can be simple and just be a parameter that only shows filename when enter inline rename
Or
-Could be parameter for request I mention in orignal thread (See "EXAMPLE" at top of page) where the extension is hidden with the first F2 press, but then becomes visible if press F2 key again.

If done this way then would not cause any issues with other users setups.
User would be able to choose to add this parameter or not.
They can also decide to setup two hotkeys one for each action.

Do not think this would cause any conflicts with the DC Option to only highlight filename and not extension when renaming.

Also with the more recently added ability to remain in inline rename mode by using up/down arrows (which I am still so thankful this was added, I use this all the time, really great addition) would just stay in the same inline rename mode that was used to enter that inline rename mode to begin with.

Thank You

**Comment by JonnyTech** [comment]
> the extension is hidden with the first F2 press, but then becomes visible if press F2 key again.

That feels very wrong. If the extension disappears, it would be confusing. No other program does that while renaming.

**Extensions are part of the filename.**

I may have files named:

```
test
test.txt
test.bmp
```

If extension disappears then which am I renaming?

And, a directory is also a filename.

If, in the example above, the first is a directory then I could alter the file structure.

DC is a tool for advanced users. Hiding extensíons is for basic users. Please do not turn Double Commander into Windows Explorer.

**Comment by ags1234** [comment]
@JonnyTech
I completely agree with "DC is a tool for advanced users. Hiding extensíons is for basic users. Please do not turn Double Commander into Windows Explorer".

However, @advice2020 asked for an option, which means he would turn _his_ DC in something he finds useful. That option would not be enabled by default, so the other users won't be affected.

**Reply by JonnyTech** [reply]
> However, advice2020 asked for an option

Indeed, but the following line implies changing the current functionality for all users:

> what about adding this to the cm_RenameOnly command that is assigned to F2 key by default.

I am not against (yet another) option getting added to DC, but realistically, if one wants that kind of behaviour then (and this is solely my opinion) maybe another file manager should be considered.

**Comment by advice2020** [comment]
Okay.... so I should find a new file manager because I do not want extensions in my inline rename field?

Tricky how to respond to these replies.
JonnyTech, you have helped me verify things in the past so I am trying not to respond back as you have just done to me.
But that does not mean that I will not point out some possible contradictory statements and possibly incorrect assumptions that have been made by an "advanced" user such as yourself :)
(Here is where I waste a lot of time)

-First off, ags1234 is correct, everything I have mentioned in this thread is presented to be added as an "Option".  I am not saying what I want should be the "default".
`cm_RenameOnly` would remain as the default that it is, I mentioned this to be added as a "New Parameter" that user would be able to decide to set if desired.

-I think DC by default uses "Name" column that does not include file extensions, but also has a Name column that does include file extensions for those who want to see them in same column.  As well as an Extension only column.
There are options for us all :)
When using the "Name" column that does NOT include extensions, then it does the opposite of what your setup does that feels "very wrong".  My column does not show extension, so it is strange to hit F2 and then all of a sudden extension appears out of nowhere.  If I am allowed to hide extensions in my Name column, then why not in my inline rename?
Now you bring up a different question, by DC offering a Name column that does NOT show extensions make it a file manager for "basic users" or would this only be if it was included in the inline renamer?
DC Multi Rename tool also splits name and extension into two separate fields.

-Correct, extensions are "part" of the filename, but when you are renaming, it is very rarely (unless txt file) "part" of the name you are going to change, therefore it can get in the way.  So yes, you and most users are used to it, but does not mean that it is convenient all the time.
Can look at it like your first name and last name.
When you are talking to people, they generally call you by just your first name because it is not convenient to have to say both each time.  But if there happens to be another Jonny in the room, then Jonny Tech may be needed :)

-If I am being referred to as a "basic user" and that you guys are "advanced" (which is probably correct)
If you have files with the same name and you put your cursor over the file you want to rename and activate F2, you do not remember what the file extension was a millisecond ago?
(With my original suggestion, another tap of the F2 key would make the extension visible to refresh your memory of what file you are in the process of renaming)
Also it is an "inline" rename so you would see the other files.
You also can have MIME type icons that differ based on extension.
If the first file in your example is a directory, there is a picture of a folder.

-So much I would like to say on this topic, but everyone always makes these old statements of Windows.
Does Windows actually allow you to hide extensions?
And if it does, DC is also on Windows, so why not make applications have similar functionality/behavior for Windows users.
Similar to how DC offers hidden setting to confirm inline rename by clicking anywhere, which is also Windows Explorer behavior.
Regardless though, the Linux system you are using right now, are you going to say that nothing about it is based on Windows?

-If want to take another Windows based statement often used.  Many say that Windows and Mac often force users to use things in a specific way, not offering much customization/settings to change things.
DC offers tons of settings/options to be useful to many users interested in customizing the file manager to their needs/use cases.
So it is strange for an "advanced" user to be against adding "yet another" option and telling someone to go use a different file manager.
One of the things that makes DC "advanced" is because it includes tons of options that users have the choice to use or not based on their needs, not just your specific needs.

Well, I am still thankful for the help you have provided me in the past, still appreciate it.

**Reply by JonnyTech** [reply]
<details>
  <summary>OFF TOPIC @advice2020:</summary>
  
> Okay.... so I should find a new file manager because I do not want extensions in my inline rename field?

No, please do not take this the wrong way. I know that when I type things, my words do seem blunt but things get lost in translation. What I mean is to find another tool to **_complement_** DC, not to **_replace_** it. For instance, just by quickly glancing at my computer, I currently use four graphical file managers, three text based file managers, eleven text editors, some of which overlap with my six development environments, four video editors, three image editors, three audio editors, four web browsers, three calculators, various messaging tools, multiple operating systems and distros, countless scripting tools and languages, in fact the more I think about tools and utilities, the list just grows and grows. I do not want all the functionality in one single tool. I want each tool to do one thing and do it well, aka the [Unix philosophy](https://en.wikipedia.org/wiki/Unix_philosophy): "Make each program do one thing well". Each application is a tool and I use that tool when I need it, in non-computing terms, I do not use a hammer with a screw, nor do I use a screwdriver with a nail, even though the end result may be similar. By adding more and more options to DC is akin to having another pocket in my shirt, soon it stops being a shirt and is just a horrible looking thing to wear - when I am working I wear one shirt, when on a date then another, and so on. So, the more "advanced" I become, the more useful utilities I find and use - and by using more tools then I find different ways of doing things that I would otherwise not discover if I use one tool made by one individual who works in a specific way. So, simply, I am suggesting to become a bit more "well travelled" and broaden your mind. And please do continue to post your thoughts and comments about improving DC but just remember that not everyone will agree with your point of view.  
</details>
